cpp/src/fil/internal.cuh

```diff
@@ -423,14 +455,16 @@ struct cat_sets_device_owner {
   cat_sets_device_owner(categorical_sets cat_sets, cudaStream_t stream)
     : bits(cat_sets.bits_size, stream), max_matching(cat_sets.max_matching_size, stream)
   {
-    if (cat_sets.max_matching != nullptr) {
+    if (cat_sets.max_matching_size > 0) {
+      ASSERT(cat_sets.max_matching != nullptr, "internal error: cat_sets.max_matching is nil");
```

Comment:
Is this assertion necessary?

Reply:
just checking for memory allocation error the first time the pointer is actually used. Not sure what's the best thing to do when out of CPU memory?